rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -213,17 +211,18 @@ object Runtime {
         shortLeashParams: ShortLeashParams[Task],
         blockTime: BlockTime[Task]
     ): RhoDispatchMap = {
+      val systemProcesses = SystemProcesses[Task](dispatcher, space)
```

Comment:
I think there's still place for improvement. Since space and dispatcher are used only in SystemProcesses then you can pass its instance to this method and remove space and dispatcher from it completely.

Reply:
Once you do that I think that new possibilities to improve the method arise. I think that it will be possible to get rid of `Task` entirely and use `F`